geode-redis/src/distributedTest/java/org/apache/geode/redis/SaddDistDunitTest.java

```diff
@@ -0,0 +1,239 @@
+  }
+
+  @Before
+  public void testSetup() {
+    jedis1.flushAll();
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    jedis1.disconnect();
+    jedis2.disconnect();
+    jedis3.disconnect();
+
+    server1.stop();
+    server2.stop();
+    server3.stop();
+  }
+
+  @Test
+  public void should_distributeDataAmongMultipleServers_givenMultipleClients() {
+
+    String key = "key";
+
+    List<String> members = makeMemberList(SET_SIZE, "member1-");
+
+    jedis1.sadd(key, members.toArray(new String[] {}));
+
+    Set<String> result = jedis2.smembers(key);
+
+    assertThat(result.toArray()).containsExactlyInAnyOrder(members.toArray());
+
+  }
+
+  @Test
+  public void should_distributeDataAmongMultipleServers_givenMultipleClients_AddingDifferentDataToSameSetConcurrently()
+      throws InterruptedException {
+
+    String key = "key";
+
+    List<String> members1 = makeMemberList(SET_SIZE, "member1-");
+    List<String> members2 = makeMemberList(SET_SIZE, "member2-");
+
+    Runnable addSetsWithClient1 = makeSADDRunnable(key, members1, jedis1);
+    Runnable addSetsWithClient2 = makeSADDRunnable(key, members2, jedis2);
+
+    runConcurrentThreads(addSetsWithClient1, addSetsWithClient2);
+
+    Set<String> results = jedis3.smembers(key);
+
+    members1.addAll(members2);
+    assertThat(results.toArray()).containsExactlyInAnyOrder(members1.toArray());
+
+  }
+
+  @Test
+  public void should_distributeDataAmongMultipleServers_givenMultipleClients_AddingSameDataToSameSetConcurrently()
+      throws InterruptedException {
+
+    String key = "key";
+
+    List<String> members = makeMemberList(SET_SIZE, "member-");
+
+    Runnable addSetsWithClient1 = makeSADDRunnable(key, members, jedis1);
+    Runnable addSetsWithClient2 = makeSADDRunnable(key, members, jedis2);
+
+    runConcurrentThreads(addSetsWithClient1, addSetsWithClient2);
+
+    Set<String> results = jedis3.smembers(key);
+
+    assertThat(results.toArray()).containsExactlyInAnyOrder(members.toArray());
+
+  }
+
+  @Test
+  public void should_distributeDataAmongMultipleServers_givenMultipleClients_AddingDifferentSetsConcurrently()
+      throws InterruptedException {
+
+    String key1 = "key1";
+    String key2 = "key2";
+
+    List<String> members1 = makeMemberList(SET_SIZE, "member1-");
+    List<String> members2 = makeMemberList(SET_SIZE, "member2-");
+
+    Runnable addSetsWithClient1 = makeSADDRunnable(key1, members1, jedis1);
+    Runnable addSetsWithClient2 = makeSADDRunnable(key2, members2, jedis2);
+
+    runConcurrentThreads(addSetsWithClient1, addSetsWithClient2);
+
+    Set<String> results1 = jedis3.smembers(key1);
+    Set<String> results2 = jedis3.smembers(key2);
+
+    assertThat(results1.toArray()).containsExactlyInAnyOrder(members1.toArray());
+    assertThat(results2.toArray()).containsExactlyInAnyOrder(members2.toArray());
+
+  }
+
+  @Test
+  public void should_distributeDataAmongMultipleServers_givenTwoSetsOfClients_OperatingOnTheSamSetConcurrently()
```

Comment:
Add a flavor of this method that has each client adding different members to the set?

Reply:
Done!